exercises/simple-linked-list/src/main/scala/SimpleLinkedList.scala

```diff
@@ -0,0 +1,10 @@
+import scala.reflect.ClassTag
+
+trait SimpleLinkedList[T] {
+  def isEmpty: Boolean
+  def value: T
+  def add(item: T): SimpleLinkedList[T]
+  def next: SimpleLinkedList[T]
+  def reverse: SimpleLinkedList[T]
+  def toArray(implicit m: ClassTag[T]): Array[T]
```

Comment:
@ricemery I think so. In the end, you could see array as an implementation detail. Why shouldn't it be a list or sequence? IMHO, each track should be able to decide for themselves which type is useful. 

There is also precendence to not using an array, as the Haskell and F# tracks only have a `toList` method (see [here](https://github.com/exercism/xhaskell/blob/master/exercises/simple-linked-list/examples/success-standard/src/LinkedList.hs) and [here](https://github.com/exercism/xfsharp/blob/master/exercises/simple-linked-list/Example.fs)).